python/ray/data/_internal/block_batching/util.py

```diff
@@ -230,19 +230,52 @@ class Sentinel:
         def __init__(self, thread_index: int):
             self.thread_index = thread_index
 
-    output_queue = queue.Queue(1)
+    queue_size = 1
```

Comment:
Could we wrap this queue functionality into a class that is a drop-in replacement for Queue? Mixing it with the application logic is going to make it hard to separate the queue vs app code in the future.

Reply:
SG, will put in a utility class.

Reply:
Refactored to a separate `Queue` class.